app/jobs/split_pdf_job.rb

```diff
@@ -45,32 +45,30 @@ def perform(exam_template, path, original_filename=nil, current_user=nil)
         user: current_user
       )
 
-      progress.total = pdf.pages.length
+      progress.total = num_pages
       partial_exams = Hash.new do |hash, key|
         hash[key] = []
       end
       num_pages_qr_scan_error = 0
-      pdf.pages.each_index do |i|
+
+      pdf.each_with_index do |page_img, i|
         split_page = SplitPage.create(filename: filename,
                                       raw_page_number: i + 1,
                                       split_pdf_log: split_pdf_log)
-        page = pdf.pages[i]
-        new_page = CombinePDF.new
-        new_page << page
-        new_page.save File.join(raw_dir, "#{split_page.id}.pdf")
+        page_img.write File.join(raw_dir, "#{split_page.id}.pdf")
+        page = CombinePDF.load(File.join(raw_dir, "#{split_page.id}.pdf"))
```

Comment:
This line probably isn't necessary any more - we shouldn't need to load the PDF here, given that we don't need it for QR code parsing below any more. So you can remove it.

However, what you *should* check (manually) is that the PDFs that are saved here are actually valid PDFs. You should download a couple from the virtual machine and check that you can open them normally in a PDF viewer.